### PR TITLE
[components-webview] Fix click handling after selection. Contributes to JB#56311

### PR DESCRIPTION
**Pull request by mkenttala**

### CODE REVIEW

File: import/controls/TextSelectionController.qml [hunk 1]
```diff
@@ -158,8 +160,20 @@ MouseArea {
         }
     }
 
+    function selectionRect() {
+        var selectionX = Math.min(start.x, end.x)
+        var selectionY = Math.min(start.y, end.y)
+        var selectionWidth = Math.max(start.x, end.x) - selectionX + end.width
+        var selectionHeight = Math.max(start.y, end.y) - selectionY + end.height
+        var rv = Qt.rect(selectionX, selectionY, selectionWidth, selectionHeight)
+        return rv
```
Comment (rainemak) [inline import/controls/TextSelectionController.qml:169]: Not quite sure why but initial size is off after entering to the text selection mode. After moving a bit left or right handle, selection rectangle updates to somewhat meaningful.

**Comment by mkenttala** [comment]
Modified to use separate rects for the start and end handles. Also made the mousearea smaller so it's a lot easier to hit the quote button in sailfish forum. Because of the animation in the handles I wasn't able to use mousearea directly so the rects are calculated whenever animations stop.

File: import/controls/TextSelectionHandle.qml [hunk 2]
```diff
@@ -53,12 +54,15 @@ Image {
         } else {
             contentItem.sendAsyncMessage("Browser:SelectionMoveEnd", selectionController.getMarkerBaseMessage(markerTag))
             moving = false
+            rect = Qt.rect(x - Theme.paddingLarge, y, width + 2 * Theme.paddingLarge, height + Theme.paddingLarge)
```
Comment (rainemak) [inline import/controls/TextSelectionHandle.qml:57]:
```suggestion
            rect = Qt.rect(mouseArea.x, mouseArea.y, mouseArea.width, mouseArea.height)
```

File: import/controls/TextSelectionHandle.qml [hunk 3]
```diff
@@ -73,9 +77,9 @@ Image {
 
         anchors {
             fill: parent
```
Comment (rainemak) [inline import/controls/TextSelectionHandle.qml:79]: Width and height above are unnecessary as there's anchors.fill.

File: import/controls/TextSelectionHandle.qml (hunk 2 above)
Comment (mkenttala) [reply import/controls/TextSelectionHandle.qml:57]: That was my first approach but it doesn't work, mouseArea only changes later and even binding to it doesn't seem to work.

File: import/controls/TextSelectionHandle.qml [hunk 4]
```diff
@@ -41,6 +41,7 @@ Image {
     property alias targetPositionAnimation: targetPositionAnimation
 
     property string markerTag
+    property rect rect: Qt.rect(0, 0, 0, 0)
```
[the remark this answers is not in the document]
Comment (mkenttala) [reply import/controls/TextSelectionHandle.qml:44]: Nope, for some reason binding to mouseArea is not working.

[the remark this answers is not in the document]
Comment (rainemak) [reply import/controls/TextSelectionHandle.qml:44]: Answering myself => no it wouldn't. Updating single rect property will not trigger update for the property itself.

Comment (rainemak) [inline import/controls/TextSelectionHandle.qml:44]: I'd avoid calling this rect, something like ```touchArea```.

File: import/controls/TextSelectionHandle.qml (hunk 2 above)
Comment (rainemak) [reply import/controls/TextSelectionHandle.qml:57]: My point is that now you have style / behaviour related values scattered here and there. 

Given no animators are used even following would be a bit cleaner:
```
    onXChanged: {
        touchArea = Qt.rect(x - Theme.paddingLarge, y, width + 2 * Theme.paddingLarge, height + Theme.paddingLarge)
    }
    onYChanged: {
        touchArea = Qt.rect(x - Theme.paddingLarge, y, width + 2 * Theme.paddingLarge, height + Theme.paddingLarge)
    }
```
Then you'd not need to care about animations and only special case would be visible false and clearing touchArea there.

**Review by rainemak — CHANGES_REQUESTED** [review]
Few minor remarks added.

File: import/controls/TextSelectionHandle.qml (hunk 2 above)
Comment (mkenttala) [reply import/controls/TextSelectionHandle.qml]: Ok, changed to that. A bit more stress for the cpu doesn't seem to affect performance.

**Review by rainemak — APPROVED** [review]
Looks good now

File: import/controls/TextSelectionHandle.qml (hunk 2 above)
Comment (rainemak) [reply import/controls/TextSelectionHandle.qml]: Yeap, that's what I checked as well.

**Pull request merged**